framework/src/constraints/MortarConstraintBase.C

```diff
@@ -81,6 +81,12 @@ MortarConstraintBase::validParams()
       "While DEFAULT quadrature order is typically sufficiently accurate, exact integration of "
       "QUAD mortar faces requires SECOND order quadrature for FIRST variables and FOURTH order "
       "quadrature for SECOND order variables.");
+  params.addParam<bool>(
+      "use_petrov_galerkin", false, "Whether to use the Petrov-Galerkin approach.");
+  params.addParam<VariableName>(
```

Comment:
```suggestion
  params.addCoupledVar(
```
This is generally necessary to ensure that variables get reinit'd properly